protocol/0087-VAMM-automated_market_maker.md

```diff
@@ -0,0 +1,165 @@
+# Automated Market Maker Framework
+
+## Summary
+
+The automated market maker (AMM) framework is designed to allow for the provision of an on-chain market making methodology which automatically provides prices according to a simple set of rules based on current market data. These rulesets are not created with the expectation of providing any profit nor of remaining solvent under any specific conditions, and so should be limited to conceptually simple setups. The initial methodology follows a concentrated-liquidity style constant-function market setup, with configurable maximum and minimum price bounds.
+
+An automated market maker is configured at a per-key level, and is enabled by submitting a transaction with the requisite parameters. At this point in time the protocol will move committed funds to a sub-account which will be used to manage margin for the AMM. Once enabled, the configuration will be added to the pool of available AMMs to be utilised by the matching engine.
+
+Each party may have only one AMM configuration per market.
+
+## Process Overview
+
+The configuration and resultant lifecycle of an automated market maker is as follows:
+
+- Party funds a key which will be used by the strategy with desired token amounts.
+- Party submits a transaction containing configuration for the strategy on a given market. This will contain:
+  - Amount of funds to commit
+  - Price bounds (upper, lower, base)
+- Additionally, the transaction should contain data related to the setup of the position but which does not need to be stored:
+  - Maximum slippage (%), used for rebasing position when creating/updating AMM
+- Once accepted, the network will transfer funds to a sub-account and use the other parameters for maintaining the position.
+- At each block, the party's available balance (including margin and general accounts) for trading on the market will be checked. If the total balance is `0` the AMM configuration will be cancelled. 
```

Comment:
Not a big thing, but usually when the network initiate cancelling something, we call it stopped, when the users does we call it cancellation, and map it to appropriate status (e.g: active, rejected (at submission), cancelled (by user), stopped (by the network).

Reply:
:+1: